index.js

```diff
@@ -63,27 +63,51 @@ exports.handler = function(event, context, callback) {
             globalCallback(null, "There's no table ID, so I'm done");
     } else {
         //go get the records we need to merge
-        getRecordsThatNeedToBeMerged(event.tableId);
+        login(event.tableId);
     }
 }
 
 
 /**
- * This function gets a tableID, finds the viewID, if one exists
- * and then grabs all the records in that view to be merged
+ * Log the user in
  * @param {Number} tableId
  */
-function getRecordsThatNeedToBeMerged(tableId){
+function login(tableId){
     //first figure out if we have a viewId associated
     //with this table Id
     var viewId = getViewForTable(tableId);
     log.log("ViewId is: " + viewId);
     //now login
-    api.login(config.account.username, config.account.password)
-    .then(() => {
-        log.log('Logged In.');
-        return api.getView(viewId, {"start": 0, "max": 1000})
-    }).then((response) =>{
+
+    //check really hard for a valid access token
+    if(config.account.access_token 
+        && typeof config.account.access_token === 'string' 
+        && config.account.access_token.length > 20){
+            log.log("Access token seems valid, using that to authorize");
+        //access token
+        api.setAccessToken(config.account.access_token);
```

Comment:
So I thought about that, to make sure login and setAccessToken are treated the same. But login requires talking to the server, whereas setting the access token is done locally in memory synchronously, so there's nothing async about the action. So I left it the way it is.